lib/Centreon/Object/Instance/Instance.php

```diff
@@ -56,4 +61,152 @@ public function getDefaultInstance()
         $row = $res->fetch();
         return $row['name'];
     }
+
+    /**
+     * Insert platform in nagios_server and platform_topology tables.
+     *
+     * @param array<string,mixed> $params
+     * @return void
+     */
+    public function insert($params = [])
+    {
+        if (!array_key_exists('ns_ip_address', $params) || !array_key_exists('name', $params)) {
+            throw new \InvalidArgumentException('Missing parameters');
+        }
+        $platformTopology = $this->findPlatformTopologyByAddress($params['ns_ip_address']);
+        $serverId = null;
+
+        if ($platformTopology !== null) {
+            if ($platformTopology->isPending() === false) {
+                throw new \Exception('Platform already created');
+            }
+
+            /**
+             * Check if the parent is a registered remote.
+             */
+            $parentPlatform = $this->findPlatformTopology($platformTopology->getParentId());
+            if ($parentPlatform !== null && $parentPlatform->getType() === PlatformRegistered::TYPE_REMOTE) {
+                if ($parentPlatform->getServerId() === null) {
+                    throw new \Exception("Parent remote server isn't registered");
+                }
+                $params['remote_id'] = $parentPlatform->getServerId();
+            }
+            $isAlreadyInTransaction = $this->db->inTransaction();
+            if (!$isAlreadyInTransaction) {
+                $this->db->beginTransaction();
+            }
+            try {
+                $serverId = parent::insert($params);
+                $platformTopology->setPending(false);
+                $platformTopology->setServerId($serverId);
+                $this->updatePlatformTopology($platformTopology);
+                $this->db->commit();
+            } catch (\Exception $ex) {
+                $this->db->rollBack();
+                throw new \Exception('Unable to update platform', 0, $ex);
+            }
+        } else {
+            $isAlreadyInTransaction = $this->db->inTransaction();
+            if (!$isAlreadyInTransaction) {
+                $this->db->beginTransaction();
+            }
+            try {
+                $serverId = parent::insert($params);
+                $params['server_id'] = $serverId;
+                $this->insertIntoPlatformTopology($params);
+                $this->db->commit();
```

Comment:
If you are already in a database transaction, the commit/rollback process should be handled by whoever started it.
```suggestion
                if (!$isAlreadyInTransaction) {
                    $this->db->commit();
                }
```